### ISSUE TITLE
Kubernetes service registry will always use a file when passed an empty string

### ISSUE THREAD

**Issue by bobbytables** [istio/istio#4244]
There's a logic flaw in this piece:
https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/client.go#L41-L50

Basically, if you pass an empty string to the resolver, it will _always_ return `username/.kube/config`.

Reason being on line 41, it checks if the string is still empty, attempts to stat and empty string (which will always fail), and then assigns the `kubeconfig` string to a statically defined kubeconfig file which doesn't exist in in-cluster configurations.

This prevents using Istio in a Kubernetes In-cluster config.

Basically using the method with an empty string always returns an error:

```
kubernetes configuration file "/root/.kube/config" does not exist
```

**Comment by kyessenov** [comment]
+1 for not doing fallback to user home. This just damaged my cluster, since I did not expect the unit tests to randomly write to my cluster.

**Comment by ZackButcher** [comment]
@costinm it looks like you made this change originally. Was there some case we hit that this code was causing problems without the attempt to fall back to the user's kubeconfig file? Otherwise yes, I think we should remove this and return an error rather than trying to use the user's config.

**Comment by ZackButcher** [comment]
Re-opening as help-wanted; this is, AFAIK, still an issue today. Shouldn't be too hard to fix.
